lib/fs.js

```diff
@@ -1794,10 +1807,17 @@ function WriteStream(path, options) {
     options = {};
   else if (typeof options === 'string')
     options = { encoding: options };
-  else if (options === null || typeof options !== 'object')
-    throw new TypeError('options must be a string or an object');
-
-  options = Object.create(options);
+  else if (options === null || typeof options !== 'object') {
+    //throw new TypeError('options must be a string or an object');
+    warnedWriteStreamOptions = internalUtil.printDeprecationMessage(
+      'Passing anything but an object or a string as the second argument to \
+WriteStream is deprecated.',
+      warnedWriteStreamOptions
+    );
+    options = options || {};
+  } else {
```

Comment:
In the ReadStream, `Object.create(options || {})` is called since 4d0329ebeb5d58702be9450e0706e0f2c404848d (v1.5.0). I don't think that there is any need to patch that nowdays and re-introduce pre-1.5.0 behavior.